sw/device/silicon_creator/lib/shutdown.c

```diff
@@ -0,0 +1,277 @@
+  }
+
+  // For each alert, read its corresponding OTP word and extract the class
+  // configuration for the current lifecycle state.
+  for (size_t i = 0; i < ALERT_HANDLER_ALERT_CLASS_MULTIREG_COUNT; ++i) {
+    uint32_t value = otp_read32(OTP_CTRL_PARAM_ROM_ALERT_CLASSIFICATION_OFFSET +
+                                i * sizeof(uint32_t));
+    alert_class_t cls = (alert_class_t)bitfield_field32_read(
+        value, (bitfield_field32_t){.mask = 0xff, .index = shift});
+    rom_error_t e = alert_configure(i, cls, enable[clsindex(cls)]);
+    if (e != kErrorOk) {
+      // Keep going if there is an error programming one alert.  We want to
+      // program them all.
+      error = e;
+    }
+  }
+
+#if 0
+  // FIXME: bring this back in after PR#6921 is merged.
+  // For each local alert, read its corresponding OTP word and extract the class
+  // configuration for the current lifecycle state.
+  for (size_t i = 0; i < ALERT_HANDLER_LOCAL_ALERT_CLASS_MULTIREG_COUNT; ++i) {
+    uint32_t value = otp_read32(OTP_CTRL_PARAM_ROM_LOCAL_ALERT_CLASSIFICATION_OFFSET +
+                                i * sizeof(uint32_t));
+    alert_class_t cls = (alert_class_t)bitfield_field32_read(
+        value, (bitfield_field32_t){.mask = 0xff, .index = shift});
+    rom_error_t e = alert_local_configure(i, cls, enable[clsindex(cls)]);
+    if (e != kErrorOk) {
+      // Keep going if there is an error programming one alert.  We want to
+      // program them all.
+      error = e;
+    }
+  }
+#endif
+
+  // For each alert class, configure the various escalation parameters.
+  for (size_t i = 0; i < ALERT_CLASSES; ++i) {
+    config.enabled = enable[i];
+    config.escalation = escalate[i];
+    config.accum_threshold = otp_read32(
+        OTP_CTRL_PARAM_ROM_ALERT_ACCUM_THRESH_OFFSET + i * sizeof(uint32_t));
+    config.timeout_cycles = otp_read32(
+        OTP_CTRL_PARAM_ROM_ALERT_TIMEOUT_CYCLES_OFFSET + i * sizeof(uint32_t));
+    for (size_t phase = 0; phase < ARRAYSIZE(config.phase_cycles); ++phase) {
+      config.phase_cycles[phase] = otp_read32(
+          OTP_CTRL_PARAM_ROM_ALERT_PHASE_CYCLES_OFFSET +
+          (i * ARRAYSIZE(config.phase_cycles) + phase) * sizeof(uint32_t));
+    }
+
+    rom_error_t e = alert_class_configure(kClasses[i], &config);
+    if (e != kErrorOk) {
+      // Keep going if there is an error programming an alert class. We want to
+      // program them all.
+      error = e;
+    }
+  }
+
+  // TODO: initialize interrupt vectors to enter shutdown_finalize.
+  return error;
+}
+
+uint32_t shutdown_redact(rom_error_t reason, shutdown_error_redact_t severity) {
+  uint32_t redacted = (uint32_t)reason;
+  if (reason == kErrorOk) {
+    return 0;
+  }
+  switch (severity) {
+    case kShutdownErrorRedactNone:
+      break;
+    case kShutdownErrorRedactError:
+      redacted &= ~kRomErrorMaskError;
+      break;
+    case kShutdownErrorRedactModule:
+      redacted &= ~(kRomErrorMaskError | kRomErrorMaskModule);
```

Comment:
Honestly I'd just clear `MaskModule` here and do a fallthrough from above.

Reply:
Done (although I use fall-through occasionally, I have doubts about its use for enhancing readability).